### PR TITLE
Feature/ifo v2

### PR DESCRIPTION
**Pull request by RabbitDoge**
IFO V2 integration (new contract + redesign)

### CODE REVIEW

File: src/hooks/ifo/v2/useGetPublicIfoData.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      const progress =
+        currentBlock > startBlockNum
+          ? ((currentBlock - startBlockNum) / totalBlocks) * 100
+          : ((currentBlock - releaseBlockNumber) / (startBlockNum - releaseBlockNumber)) * 100
+
+      setState({
+        secondsUntilEnd: blocksRemaining * BSC_BLOCK_TIME,
+        secondsUntilStart: (startBlockNum - currentBlock) * BSC_BLOCK_TIME,
+        poolBasic: { ...formatPool(poolBasic), taxRate: 0 },
+        poolUnlimited: { ...formatPool(poolUnlimited), taxRate: taxRate / 1000000 },
```
Comment (ChefNyan) [inline src/hooks/ifo/v2/useGetPublicIfoData.ts:101]: It needs to be adjusted

File: src/utils/formatBalance.ts [hunk 2]
```diff
@@ -8,3 +8,11 @@ export const getBalanceNumber = (balance: BigNumber, decimals = 18) => {
...
   return balance.dividedBy(new BigNumber(10).pow(decimals)).toFixed()
 }
+
+export const formatNumber = (number: number, minPrecision = 2, maxPrecision = 2) => {
+  const options = {
+    minimumFractionDigits: minPrecision,
+    maximumFractionDigits: maxPrecision,
+  }
+  return number.toLocaleString(undefined, options)
+}
```
Comment (hachiojidev) [inline src/utils/formatBalance.ts:18]: Nice

File: src/views/Ifos/components/IfoV1Card/ContributeModal.tsx [hunk 3]
```diff
@@ -3,13 +3,13 @@ import { useWeb3React } from '@web3-react/core'
...
 import { ethers } from 'ethers'
 import { Modal, LinkExternal, Box, Text } from '@pancakeswap-libs/uikit'
-import BalanceInput from 'components/BalanceInput'
 import useTokenBalance from 'hooks/useTokenBalance'
 import { getBalanceNumber } from 'utils/formatBalance'
 import useI18n from 'hooks/useI18n'
 import ApproveConfirmButtons from 'views/Profile/components/ApproveConfirmButtons'
 import useApproveConfirmTransaction from 'hooks/useApproveConfirmTransaction'
 import { useERC20 } from 'hooks/useContract'
+import BalanceInput from './BalanceInput'
```
Comment (hachiojidev) [inline src/views/Ifos/components/IfoV1Card/ContributeModal.tsx:12]: Is this the same as the one in the UI Kit?

Comment (RabbitDoge) [reply src/views/Ifos/components/IfoV1Card/ContributeModal.tsx:12]: Yes. It's the only place where it's still used. After the release, everything that is inside the V1 folder will be removed

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**